File: ChatClientCS/Models/Participant.cs
using ChatClientCS.ViewModels;
using System.Collections.ObjectModel;

namespace ChatClientCS.Models
{
    public class Participant : ViewModelBase
    {
        public string Name { get; set; }
        public byte[] Photo { get; set; }
        public ObservableCollection<ChatMessage> Chatter { get; set; }

        private bool _isLoggedIn = true;
        public bool IsLoggedIn
        {
            get { return _isLoggedIn; }
            set { _isLoggedIn = value; OnPropertyChanged(); }
        }

        private bool _hasSentNewMessage;
        public bool HasSentNewMessage
        {
            get { return _hasSentNewMessage; }
            set { _hasSentNewMessage = value; OnPropertyChanged(); }
        }

        private bool _isTyping;
        public bool IsTyping
        {
            get { return _isTyping; }
            set { _isTyping = value; OnPropertyChanged(); }
        }

        public Participant() { Chatter = new ObservableCollection<ChatMessage>(); }
    }
}

File: ChatClientVB/Models/Participant.vb
Imports System.Collections.ObjectModel

Public Class Participant
    Inherits ViewModelBase

    Public Property Name As String
    Public Property Photo As Byte()
    Public Property Chatter As New ObservableCollection(Of ChatMessage)

    Private _isLoggedIn As Boolean = True
    Public Property IsLoggedIn As Boolean
        Get
            Return _isLoggedIn
        End Get
        Set(value As Boolean)
            _isLoggedIn = value
            OnPropertyChanged()
        End Set
    End Property

    Private _hasSentNewMessage As Boolean
    Public Property HasSentNewMessage As Boolean
        Get
            Return _hasSentNewMessage
        End Get
        Set(value As Boolean)
            _hasSentNewMessage = value
            OnPropertyChanged()
        End Set
    End Property

    Private _isTyping As Boolean
    Public Property IsTyping As Boolean
        Get
            Return _isTyping
        End Get
        Set(value As Boolean)
            _isTyping = value
            OnPropertyChanged()
        End Set
    End Property
End Class
